### PR TITLE
Improve tracing

### PR DESCRIPTION
**Pull request by bonclay7**
### What does this PR do?

This PR provides improvements to the tracing config by removing the daemonset mode when tracing is enabled. By default `enable_tracing` is set to `true`. Meaning receivers and exporters will be configured to send send OTLP traces and metrics to Amazon Managed Prometheus and AWS X-Ray respectively.

Provides as well documentation for a tracing example, this depends on https://github.com/aws-observability/aws-otel-community/pull/586 to be merged

ADOT log level fixes: using new verbosity level semantics

### More

- [x] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)
- [x] Yes, I ran `pre-commit run -a` with this PR
- [x] Yes, I have added a new example under [examples](https://github.com/aws-observability/terraform-aws-observability-accelerator/tree/main/examples) to support my PR (when applicable)
- [x] Yes, I have updated the [Pages](https://github.com/aws-observability/terraform-aws-observability-accelerator/tree/main/docs) for this feature

**Note**: Not all the PRs required examples and docs.

### For Moderators
- [x] E2E Test successfully complete before merge?

### CODE REVIEW

**Comment by bonclay7** [comment]
@lewinkedrs this is ready for review

File: docs/eks/tracing.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+
+=== "using docker"
+
+    ``` console
+    export ECR_REPOSITORY_URI=$(aws ecr create-repository --repository go-sample-app --query repository.repositoryUri --output text)
+    aws ecr get-login-password --region $AWS_REGION | docker login --username AWS --password-stdin 339743103717.dkr.ecr.eu-central-1.amazonaws.com
+    docker tag go-sample-app:latest "${ECR_REPOSITORY_URI}:latest"
+    docker push "${ECR_REPOSITORY_URI}:latest"
+    ```
```
Comment (lewinkedrs) [inline docs/eks/tracing.md:73]: i think this is a reference to your ECR:
```
--password-stdin 339743103717.dkr.ecr.eu-central-1.amazonaws.com

```
I would write something like $PATHTOECR and then let the user fill that in with what they need.

File: docs/eks/tracing.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+spec:
+  type: ClusterIP
+  selector:
+    app: go-sample-app
+  ports:
+    - protocol: TCP
+      port: 8080
+      targetPort: 8080
+```
+
```
Comment (lewinkedrs) [inline docs/eks/tracing.md:145]: I find this section a little confusing. I would say lets remove the bit about cat and EOF and just say like "Create a file called 'eks.yaml' with the following contents".

File: docs/eks/tracing.md (hunk 1 above)
Comment (bonclay7) [reply docs/eks/tracing.md:73]: oh yeah definitely not right, actually the $ECR_REPOSITORY_URI should do it, will fix

File: docs/eks/tracing.md (hunk 2 above)
Comment (bonclay7) [reply docs/eks/tracing.md:141]: done

File: docs/eks/tracing.md (hunk 1 above)
Comment (bonclay7) [reply docs/eks/tracing.md:73]: fixed

**Review by lewinkedrs — APPROVED** [review]
LGTM

**Comment by bonclay7** [comment]
queued up for #202

**Pull request merged**